libavfilter/vf_vpp_qsv.c

```diff
@@ -460,11 +486,27 @@ static int config_output(AVFilterLink *outlink)
 #endif
     }
 
+    if (vpp->scaling_mode) {
+#ifdef QSV_HAVE_SCALING
+        memset(&vpp->scaling_conf, 0, sizeof(mfxExtVPPScaling));
+        vpp->scaling_conf.Header.BufferId = MFX_EXTBUFF_VPP_SCALING;
+        vpp->scaling_conf.Header.BufferSz = sizeof(mfxExtVPPScaling);
+        vpp->scaling_conf.ScalingMode = vpp->scaling_mode;
+
+        param.ext_buf[param.num_ext_buf++] = (mfxExtBuffer*)&vpp->scaling_conf;
+#else
+        av_log(ctx, AV_LOG_WARNING, "The scaling_mode option is "
+            "not supported with this MSDK version.\n");
+        vpp->scaling_mode = MFX_SCALING_MODE_DEFAULT;
```

Comment:
is MFX_SCALING_MODE_DEFAULT defined in #else branch?

Reply:
Thanks for catching this. The doc didn't mention when MFX_SCALING_MODE_DEFAULT was added. But according to https://github.com/Intel-Media-SDK/MediaSDK/blob/master/doc/mediasdk-man.md, MFX_EXTBUFF_VPP_SCALING was added in the SDK API 1.19. I guess MFX_SCALING_MODE_DEFAULT was added in 1.19 too.

Reply:
Updated  in the new version, thanks.